packages/client/tests/functional/referential-integrity/tests.ts

```diff
@@ -0,0 +1,677 @@
+        })
+        expect(profiles).toEqual([
+          {
+            id: '2',
+            userId: '2',
+          },
+          {
+            id: '3',
+            userId: '1',
+          },
+        ])
+      })
+
+      test('[update] child id with existing id should throw', async () => {
+        await prisma[userModel].create({
+          data: {
+            id: '1',
+            profile: {
+              create: { id: '1' },
+            },
+          },
+        })
+        await prisma[userModel].create({
+          data: {
+            id: '2',
+            profile: {
+              create: { id: '2' },
+            },
+          },
+        })
+
+        await expect(
+          prisma[profileModel].update({
+            where: { id: '1' },
+            data: {
+              id: '2', // existing id
+            },
+          }),
+        ).rejects.toThrowError('Unique constraint failed on the fields: (`id`)')
+      })
+
+      test('[updateMany] parent id should succeed', async () => {
+        await prisma[userModel].create({
+          data: {
+            id: '1',
+            profile: {
+              create: { id: '1' },
+            },
+          },
+        })
+        await prisma[userModel].create({
+          data: {
+            id: '2',
+            profile: {
+              create: { id: '2' },
+            },
+          },
+        })
+
+        await prisma[userModel].updateMany({
+          data: { id: '3' },
+          where: { id: '1' },
+        })
+
+        expect(
+          prisma[userModel].findUnique({
+            where: { id: '1' },
+          }),
+        ).resolves.toEqual(null)
+
+        const user1WithNewId = await prisma[userModel].findUniqueOrThrow({
+          where: { id: '3' },
+          include: { profile: true },
+        })
+        expect(user1WithNewId).toEqual({
+          id: '3',
+          profile: {
+            id: '1',
+            userId: '3',
+          },
+        })
+
+        const profile1FromUser3 = await prisma[profileModel].findUniqueOrThrow({
+          where: { userId: '3' },
+        })
+        expect(profile1FromUser3).toEqual({
+          id: '1',
+          userId: '3',
+        })
+      })
+
+      test('[updateMany] parent id with existing id should throw', async () => {
+        await prisma[userModel].create({
+          data: {
+            id: '1',
+            profile: {
+              create: { id: '1' },
+            },
+          },
+        })
+        await prisma[userModel].create({
+          data: {
+            id: '2',
+            profile: {
+              create: { id: '2' },
+            },
+          },
+        })
+
+        try {
+          await prisma[userModel].updateMany({
+            data: { id: '2' }, // existing id
+            where: { id: '1' },
+          })
+          assertUnreachable()
+        } catch (error) {
+          expect(error.message).toContain('Unique constraint failed on the fields: (\`id\`)')
+        }
```

Comment:
Can be simplified to
```suggestion
        await expect(
          prisma[userModel].updateMany({
            where: { id: '1' },
            data: {
              id: '2', // existing id
            },
          }),
        ).rejects.toThrowError('Unique constraint failed on the fields: (`id`)')
```